ouroboros-network/sim-tests-lib/Test/Ouroboros/Network/Testnet.hs

```diff
@@ -828,15 +828,17 @@ prop_track_coolingToCold_demotions defaultBearerInfo diffScript =
           decreasing [_] = True
           decreasing (x : y : t)
             | x > y     = decreasing (y : t)
+            | x == 0    = decreasing (y : t)
```

Comment:
This means we allow the set to increase its size from empty to non-empty, I'd add a comment.

Reply:
I actually don't understand why `inProgressDemoteToCold` should only shrink in general? Is this test only applied to specific arguments / scenario?

Reply:
This test checks that the node doesn't keep any peers stuck in `inProgressDemoteToCold`. When a node in the testnet disconnects it is going to be added to the `inProgressDemoteToCold` set and the set will only grow after we are able to connect to it again and it disconnects again. We kind of rely on the sequential nature of IOSim but I couldn't find a simpler way to write the test

Reply:
The fix here makes sure that we ignore when the set goes to 0 elements and then grows back again.

Reply:
> This test checks that the node doesn't keep any peers stuck in inProgressDemoteToCold.

If that's what we want to test, maybe we should redesign the test to make sure none of the peers stays in the `inProgressDemoteToCold` for too long.  Is this expressible with the events API?

Reply:
It is definitely more tricky, but I believe it could be done

Reply:
Could you create & label an issue for this.